tower/src/retry/policy.rs

```diff
@@ -58,7 +58,7 @@ pub trait Policy<Req, Res, E>: Sized {
     ///
     /// [`Service::Response`]: crate::Service::Response
     /// [`Service::Error`]: crate::Service::Error
-    fn retry(&self, req: &Req, result: Result<&Res, &E>) -> Option<Self::Future>;
+    fn retry(&self, req: &mut Req, result: &mut Result<Res, E>) -> Option<Self::Future>;
```

Comment:
I guess getting a `&mut Result<Res, E>` allows users to change `Err(_)` into `Ok(_)` which might be kinda odd. Not sure. Feels like getting `Result<&mut Res, &mut E>` would fit the use-cases as well but not allow changing the `Result` variant.

Reply:
comment wasn't threaded because of email:
> The entire point is to change the variant (at least for me). Because there
are "success" cases the actually need to be retries and when I run out of
retries I need to change it to failure